### PR TITLE
Add templates

### PR DESCRIPTION
**Pull request by QuinnNgo**
# Description
This PR adds an `opta init` command, which initializes opta files for users. Right now, there are two templates for environments: `aws` and `gcp`, and only one template for services: `k8s`.

# Screenshot: Example run
image

### CODE REVIEW

File: opta/commands/init.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+Press ^C at any time to quit.
+    """
+    )
+
+    env_dict: Optional[dict] = None
+
+    template = SERVICE_TEMPLATES["k8s"]
+    res = template.run()
+
+    try:
```
Comment (QuinnNgo) [inline opta/commands/init.py:101]: TODO: make env file required here

File: opta/commands/init_templates/environment/aws/opta.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+org_name: ""
+providers:
+  aws:
+    region: ""
+    account_id: ""
+modules:
+  - type: base
+  - type: dns
+    delegated: false
+  - name: deployeruser
```
Comment (juandiegopalomino) [inline opta/commands/init_templates/environment/aws/opta.yaml:11]: don't need this-- this creates an admin user. We can start prompting folks to create users later

File: opta/commands/init_templates/environment/aws/opta.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    type: aws-iam-user
+    extra_iam_policies:
+      - "arn:aws:iam::aws:policy/AdministratorAccess"
+  - name: deployerrole
```
Comment (juandiegopalomino) [inline opta/commands/init_templates/environment/aws/opta.yaml:15]: Don't need this

File: opta/commands/init_templates/environment/aws/opta.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    type: aws-iam-role
+    extra_iam_policies:
+      - "arn:aws:iam::aws:policy/AdministratorAccess"
+    allowed_iams:
+      - "${{module.deployeruser.user_arn}}"
+  - type: k8s-cluster
+    max_nodes: 12
+  - type: k8s-base
+    admin_arns:
+      - "${{module.deployeruser.user_arn}}"
```
Comment (juandiegopalomino) [inline opta/commands/init_templates/environment/aws/opta.yaml:25]: you'll want to remove this too

**Review by juandiegopalomino — COMMENTED** [review]
could you add some more description to the different prompts? Maybe even links

File: opta/commands/init_templates/environment/aws/opta.yaml (hunk 2 above)
Comment (QuinnNgo) [reply opta/commands/init_templates/environment/aws/opta.yaml:11]: Ah, thank you for pointing this out!

**Comment by QuinnNgo** [comment]
> could you add some more description to the different prompts? Maybe even links

What exactly did you have in mind? Links to what?

File: opta/commands/init_templates/environment/aws/opta.yaml (hunk 2 above)
Comment (juandiegopalomino) [reply opta/commands/init_templates/environment/aws/opta.yaml:11]: then pls remove

File: opta/commands/init_templates/environment/aws/opta.yaml (hunk 3 above)
Comment (juandiegopalomino) [reply opta/commands/init_templates/environment/aws/opta.yaml:15]: ping

File: opta/commands/init_templates/variables/aws_account_id.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    set_path = dictionary_deep_set(["providers", "aws", "account_id"])
+    set_path(d, v)
+    return d
+
+
+accountIdVariable = TemplateVariable(
+    prompt="aws account id",
+    applier=apply,
+    validator=validate,
+    error_message="invalid account id",
```
Comment (juandiegopalomino) [inline opta/commands/init_templates/variables/aws_account_id.py:21]: "Invalid account id: must be of the form blah blah blah"

File: opta/commands/init_templates/variables/aws_region.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    set_path = dictionary_deep_set(["providers", "aws", "region"])
+    set_path(d, v)
+    return d
+
+
+indented_regions = [f"\t{region}" for region in REGIONS]
+region_string = "\n".join(indented_regions)
+
+awsRegionVariable = TemplateVariable(
+    prompt="aws region",
```
Comment (juandiegopalomino) [inline opta/commands/init_templates/variables/aws_region.py:45]: "Select your AWS region https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html"

File: opta/commands/init_templates/variables/dns_domain.py [hunk 7]
```diff
@@ -0,0 +1,6 @@
+from opta.commands.init_templates.helpers import set_module_field
+from opta.commands.init_templates.template import TemplateVariable
+
+dnsDomainVariable = TemplateVariable(
+    prompt="dns domain", applier=set_module_field("dns", ["domain"]),
```
Comment (juandiegopalomino) [inline opta/commands/init_templates/variables/dns_domain.py:5]: "Please specify the dns domain you wish to use (e.g. mydomain.dev)"

File: opta/commands/init_templates/environment/aws/opta.yaml (hunk 2 above)
Comment (QuinnNgo) [reply opta/commands/init_templates/environment/aws/opta.yaml:11]: whoops -- didn't realize i didn't push

Comment (QuinnNgo) [reply opta/commands/init_templates/environment/aws/opta.yaml]: c61a0a2f4b017d0cadd2fbe8070cc0f0e94d33ec

**Review by juandiegopalomino — COMMENTED** [review]
capitalize AWS and GCP and first letter of each prompt

**Comment by QuinnNgo** [comment]
> capitalize AWS and GCP and first letter of each prompt

d9a4d35a080e327d4892f2a6a1accaefbca3231f

**Review by juandiegopalomino — APPROVED** [review]

**Pull request merged**